src/internal.c

```diff
@@ -15298,43 +15301,40 @@ static void DtlsUpdateWindowGTSeq(word32 diff, word32* window)
             else {
                 temp |= (oldWindow[i-idx] << newDiff);
                 window[i] = temp;
-                temp = oldWindow[i-idx] >> (DTLS_WORD_BITS - newDiff - 1);
+                temp = oldWindow[i-idx] >> (DTLS_WORD_BITS - newDiff);
             }
         }
     }
     window[0] |= 1;
 }
 
-static WC_INLINE int _DtlsUpdateWindow(WOLFSSL* ssl, word16* next_hi,
-                                       word32* next_lo, word32 *window)
+static WC_INLINE int _DtlsUpdateWindow(word16 cur_hi, word32 cur_lo,
+                                       word16* next_hi, word32* next_lo,
+                                       word32 *window)
 {
-    word32 cur_lo, diff;
+    word32 diff;
     int curLT;
-    word16 cur_hi;
-
-    cur_hi = ssl->keys.curSeq_hi;
-    cur_lo = ssl->keys.curSeq_lo;
 
     if (cur_hi == *next_hi) {
         curLT = cur_lo < *next_lo;
-        diff = curLT ? *next_lo - cur_lo - 1 : cur_lo - *next_lo + 1;
+        diff = curLT ? *next_lo - cur_lo : cur_lo - *next_lo;
     }
     else {
         curLT = cur_hi < *next_hi;
-        diff = curLT ? cur_lo - *next_lo - 1 : *next_lo - cur_lo + 1;
+        diff = curLT ? cur_lo - *next_lo : *next_lo - cur_lo;
     }
 
     if (curLT) {
-        word32 idx = diff / DTLS_WORD_BITS;
-        word32 newDiff = diff % DTLS_WORD_BITS;
+        word32 idx = (diff - 1) / DTLS_WORD_BITS;
+        word32 newDiff = (diff - 1) % DTLS_WORD_BITS;
```

Comment:
should this be checked against underflow?

Reply:
When you get to this location, diff - 1 > 0.